libs/remix-ui/terminal/src/lib/remix-ui-terminal.tsx

```diff
@@ -0,0 +1,551 @@
+        }
+      }
+    }
+    if (event.which === 13 && !autoCompletState.showSuggestions) {
+      if (event.ctrlKey) { // <ctrl+enter>
+        // on enter, append the value in the cli input to the journal
+        inputEl.current.focus()
+      } else { // <enter>
+        event.preventDefault()
+        setCmdIndex(-1)
+        setCmdTemp('')
+        const script = autoCompletState.userInput.trim() // inputEl.current.innerText.trim()
+        if (script.length) {
+          cmdHistoryDispatch({ type: 'cmdHistory', payload: { script } })
+          newstate.commands.script(wrapScript(script))
+        }
+        setAutoCompleteState(prevState => ({ ...prevState, userInput: '' }))
+        inputEl.current.innerText = ''
+        inputEl.current.focus()
+        setAutoCompleteState(prevState => ({ ...prevState, showSuggestions: false }))
+      }
+    } else if (newstate._commandHistory.length && event.which === 38 && !autoCompletState.showSuggestions && (autoCompletState.userInput === '')) {
+      event.preventDefault()
+      setAutoCompleteState(prevState => ({ ...prevState, userInput: newstate._commandHistory[0] }))
+    } else if (event.which === 38 && autoCompletState.showSuggestions) {
+      event.preventDefault()
+      if (autoCompletState.activeSuggestion === 0) {
+        return
+      }
+      setAutoCompleteState(prevState => ({ ...prevState, activeSuggestion: suggestionCount - 1, userInput: Object.keys(autoCompletState.data._options[autoCompletState.activeSuggestion]).toString() }))
+    } else if (event.which === 38 && !autoCompletState.showSuggestions) { // <arrowUp>
+      if (cmdHistory.length - 1 > _cmdIndex) {
+        setCmdIndex(prevState => prevState++)
+      }
+      inputEl.current.innerText = cmdHistory[_cmdIndex]
+      inputEl.current.focus()
+    } else if (event.which === 40 && autoCompletState.showSuggestions) {
+      event.preventDefault()
+      if ((autoCompletState.activeSuggestion + 1) === autoCompletState.data._options.length) {
+        return
+      }
+      setAutoCompleteState(prevState => ({ ...prevState, activeSuggestion: suggestionCount + 1, userInput: Object.keys(autoCompletState.data._options[autoCompletState.activeSuggestion + 1]).toString() }))
+    } else if (event.which === 40 && !autoCompletState.showSuggestions) {
+      if (_cmdIndex > -1) {
+        setCmdIndex(prevState => prevState--)
+      }
+      inputEl.current.innerText = _cmdIndex >= 0 ? cmdHistory[_cmdIndex] : _cmdTemp
+      inputEl.current.focus()
+    } else {
+      setCmdTemp(inputEl.current.innerText)
+    }
+  }
+
+  /* start of mouse events */
+
+  const mousedown = (event: MouseEvent) => {
+    setSeparatorYPosition(event.clientY)
+    const leftRef = document.getElementById('dragId')
```

Comment:
Why not use useRef hook instead of getElementById. Or manipulate styling by state change directly on element?